aten/src/ATen/native/Pow.cpp

```diff
@@ -6,85 +6,64 @@
 #include <ATen/ScalarOps.h>
 #include <ATen/native/Resize.h>
 
-namespace at { namespace native {
+namespace at {
+namespace meta {
 
-DEFINE_DISPATCH(pow_tensor_tensor_stub);
-DEFINE_DISPATCH(pow_tensor_scalar_stub);
-
-Tensor& pow_out(Tensor& result, const Tensor& base, const Tensor& exp) {
-  if (exp.dim() == 0 && exp.device().is_cpu() && base.is_cuda()) {
-    return native::pow_out(result, base, exp.item());
-  }
-  auto iter = TensorIterator::binary_op(result, base, exp);
-  pow_tensor_tensor_stub(iter.device_type(), iter);
-  return result;
+TORCH_META_FUNC2(pow, Tensor_Tensor) (const Tensor& base, const Tensor& exp) {
+  build_binary_op(maybe_get_output(), base, exp);
 }
 
-Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
+TORCH_META_FUNC2(pow, Tensor_Scalar) (const Tensor& base, const Scalar exp) {
   // Numpy compatibility check:
   TORCH_CHECK(!(isIntegralType(base.scalar_type(), true) &&
               exp.isIntegral(true) && exp.toLong() < 0),
               "Integers to negative integer powers are not allowed.");
 
   auto common_dtype = at::result_type(base, exp);
-  TORCH_CHECK(at::can_cast(common_dtype, result.scalar_type()),
-           "result type ", common_dtype, " can't be cast to the desired output type ",
-           result.scalar_type());
```

Comment:
So, before the port this check ran before TensorIterator creation. After the port though, it would have to live inside of the `TORCH_IMPL_FUNC` block, since it directly uses the output tensor. I ended up removing it because:
- It didn't appear to get hit anywhere in tests- the tests that used to hit this bit of code now fail with an assert from inside of TensorIterator first.
- It seems preferable to not have any dtype/shape asserts happen in the impl() call. Preferably, they should all happen inside the meta call, and this seems like a case that TensorIterator handles.